utf8/reader_test.go

```diff
@@ -0,0 +1,288 @@
+package utf8_test
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	stdiotest "testing/iotest"
+
+	"github.com/madlambda/spells/assert"
+	"github.com/madlambda/spells/io/runes"
+	"github.com/madlambda/spells/iotest"
+	"github.com/madlambda/spells/utf8"
+)
+
+const expectedString = "Εν οίδα ότι ουδέν οίδα"
+
+func TestUTF8Reader(t *testing.T) {
+	type testcase struct {
+		input  string
+		repeat int
+		err    error
+		errOut []rune
+	}
+
+	testcases := []testcase{
```

Comment:
Maybe I'm just missing that this is already tested here, but I was thinking on having also test cases for:

- Reading single char that is formed by 2 bytes
- Reading single char that is formed by 3 bytes
- Reading single char that is formed by 4 bytes

And then doing the same but for a string that has single byte ASCII chars but that starts/ends with a multi-byte char (for each size). But maybe this is already covered, so disregard on that case.

Reply:
I added the testcases, have a look.